src/x/debug/pprof.go

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) 2021  Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+// Package debug contains debug helpers for M3 components.
+package debug
+
+import (
+	"net/http"
+	"net/http/pprof"
```

Comment:
Note, this will still technically register pprof routes on the DefaultServeMux as well since there's an `init` function in the package that's responsible for this. This ends up having no impact as 1) no server is using the DefaultServeMux and 2) init() functions are only executed the first time a package is loaded so it won't fail integration tests restarting M3 components.

Reply:
SG

Reply:
can we double check pprof still works as expected in a real env?

Reply:
Good call. Everything still looks good!

```
nb@nb monorepo (nb/pprof-test) (⎈ |rc:rc-06) $ ./bin/droidcli debug pprof -e rc_rc_06 m3agg-primary-rep0-0
2021-09-12T16:29:59.450-0400	INFO	performing profile	{"url": "http://localhost:52806/debug/pprof/heap"}
2021-09-12T16:29:59.690-0400	INFO	writing profile to disk	{"filename": "pprof.heap.1631478597.out"}

nb@nb monorepo (nb/pprof-test) (⎈ |rc:rc-06) $ ./bin/droidcli debug pprof -e rc_rc_06 m3coordinator-read-cfd89d8d6-88ztt
2021-09-12T17:03:57.759-0400	INFO	performing profile	{"url": "http://localhost:55678/debug/pprof/heap"}
2021-09-12T17:03:57.962-0400	INFO	writing profile to disk	{"filename": "pprof.heap.1631480635.out"}

nb@nb monorepo (nb/pprof-test) (⎈ |rc:rc-06) $ ./bin/droidcli debug pprof -e rc_rc_06 m3-cluster-short-rep0-0
2021-09-12T17:42:07.923-0400	INFO	performing profile	{"url": "http://localhost:57454/debug/pprof/heap"}
2021-09-12T17:42:08.092-0400	INFO	writing profile to disk	{"filename": "pprof.heap.1631482926.out"}

nb@nb monorepo (nb/pprof-test) (⎈ |rc:rc-06) $ ./bin/droidcli debug dump -C master -e rc_rc_06 m3-cluster-short-rep0-0
2021-09-12T17:43:04.509-0400	INFO	performing profile	{"url": "http://localhost:57525/debug/dump"}
2021-09-12T17:43:09.970-0400	INFO	writing profile to disk	{"filename": "pprof.dump.1631482982.zip"}
```